client/can/adapters.py

```diff
@@ -135,8 +136,7 @@ def send_frame(self, frame):
     def receive_frame(self):
         try:
             # according to the datasheet, erasing a sector from an stm32f407
-            # can take up to 4 seconds. Therefore wait for 5 seconds before
-            # assuming nobody answered.
-            return self.rx_queue.get(True, 5)
+            # can take up to 4 seconds. Therefore we need to wait a bit
+            return self.rx_queue.get(True, self.timeout)
```

Comment:
Please also update the socket CAN timeout above: `self.socket.settimeout(timeout)`

Reply:
It's not enough to get with timeout?

Reply:
I updated the SocketCAN code above, but I dont have a way to test it.

> It's not enough to get with timeout?

Nope, the queue is only used when in SLCAN mode.